osclib/accept_command.py

```diff
@@ -128,20 +179,42 @@ def cleanup(self, project):
 
     def accept_other_new(self):
         changed = False
+
+        if self.api.delreq_review:
+            rqlist = self.find_virtually_accepted_requests(self.api.project)
+            for req in rqlist:
+                if self.virtual_accept_request_has_no_binary(self.api.project, req['packages'][0]):
+                    # Accepting delreq-review review
+                    self.api.do_change_review_state(req['id'], 'accepted',
+                                                    by_group=self.api.delreq_review,
+                                                    message='Virtually accepted delete {}'.format(req['packages'][0]))
+
         rqlist = self.find_new_requests(self.api.project)
         if self.api.cnonfree:
             rqlist += self.find_new_requests(self.api.cnonfree)
 
         for req in rqlist:
             oldspecs = self.api.get_filelist_for_package(pkgname=req['packages'][0], project=self.api.project, extension='spec')
             print 'Accepting request %d: %s' % (req['id'], ','.join(req['packages']))
+            if req['type'] == 'delete':
+                # Remove devel project/package tag before accepting the request
+                self.remove_obsoleted_develtag(self.api.project, req['packages'][0])
             change_request_state(self.api.apiurl, str(req['id']), 'accepted', message='Accept to %s' % self.api.project)
             # Check if all .spec files of the package we just accepted has a package container to build
             self.create_new_links(self.api.project, req['packages'][0], oldspecs)
             changed = True
 
         return changed
 
+    def remove_obsoleted_develtag(self, project, package):
+        xpath = {
+            'package': "@project='%s' and devel/@project='%s' and devel/@package='%s'" % (project, project, package),
```

Comment:
This removes the sub-packages which have their devel project set to the target project right? Not sure if you like, but one tweak that I checked works is:

```
@project='%s' and devel/@project=@project and devel/@package='%s'
```

to avoid adding project twice.

Reply:
No, it only removes the devel attribute from the packages - it does not remove the package (but as long as the devel attribute is set, accepting the delete request fails; with or without the virtual delete

Reply:
It only removes the devel tag from the package if it has set devel pacakge in the same project.

Reply:
Yes, that's what I meant. Obviously, deleting a devel project would be a bit crazy.